### PR TITLE
[NP-8596] FOAMException consolidation, Toast notification title support. 

### PR DESCRIPTION
**Pull request by jlhughes**
- Consolidate Exception interface and ExceptionInterface into FOAMException
- provide support for override 'title' of Notification toast messages

### CODE REVIEW

**Review by lchanmann — COMMENTED** [review]
So if FOAMException will be thrown to the client then do we still need ClientRuntimeException?

**Comment by kgrgreer** [comment]
LGTM

**Comment by jlhughes** [comment]
> So if FOAMException will be thrown to the client then do we still need ClientRuntimeException?

Good question. I'm **not** completely sure of the interaction between ClientRuntimeException and flag isClientException. 
For now, to avoid subtle breaks, I've set isClientException in FOAMException to false.  ClientRuntimeException extends FOAMException and sets isClientException to true.  And existing exceptions such AuthenticationException continue to extend ClientRuntimeException.

**Review by lchanmann — APPROVED** [review]

File: src/foam/u2/dialog/NotificationMessage.js [hunk 1]
```diff
@@ -146,12 +146,16 @@ foam.CLASS({
...
-          if ( this.message.endsWith('Exception') ) {
-            this.message = this.message.replace('Exception', '');
+          if ( ex.title ) {
+            this.message = ex.title;
+          } else {
+            this.message = ex.id.split('.').pop();
+            if ( this.message.endsWith('Exception') ) {
+              this.message = this.message.replace('Exception', '');
+            }
+            this.message = foam.String.capitalize(foam.String.labelize(this.message).toLowerCase());
```
Comment (sarthak-marwaha) [inline src/foam/u2/dialog/NotificationMessage.js:156]: Why do we labelize it when we end up capitalizing it?

**Pull request merged**